common/Protocol.hpp

```diff
@@ -101,6 +101,18 @@ namespace COOLProtocol
         return false;
     }
 
+    template <std::size_t N>
+    inline bool getNonNegTokenInteger(const std::string& token, const char (&name)[N], int& value)
+    {
+        bool ret = getTokenInteger(token, name, value);
+        if (!ret)
+        {
+            return ret;
+        }
+
+        return value >= 0;
```

Comment:
I guess if we want to be more terse we could put it on one line:

`return getTokenInteger(token, name, value) && value >= 0;`

But this is just a personal preference. Your version is perfectly fine, of course.

Reply:
Sure, why not.